### PR TITLE
Add support for overworld sprite gender differences + add all the sprites

### LINKED ISSUE #5124: Pyroar and Indede don't use female follower sprites
### Description

Title 
IMG_20240809_012012
IMG_20240809_011936
Screenshot_2024-08-09-01-20-23-022_it dbtecno pizzaboygba-edit

Screenshot_2024-08-09-01-48-43-197_it dbtecno pizzaboygba-edit
IMG_20240809_020309
Screenshot_2024-08-09-01-48-10-752_it dbtecno pizzaboygba-edit

### Version

1.9.0 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by kittenchilly**
## Description
Allows overworld sprites and followers to have gender differences defined.
Adds sprites for all current pokemon gender differences.

## Issue(s) that this PR fixes
Fixes #5124 

## **People who collaborated with me in this PR**
All the sprites for gender differences that weren't from GF came from here: https://www.deviantart.com/larryturbo/art/OWs-of-Missing-Gender-Variants-Normal-Shiny-878691887

## **Discord contact info**
kittenchilly

### CODE REVIEW

File: src/data/pokemon/species_info/gen_1_families.h [hunk 1]
```diff
@@ -1874,6 +1886,12 @@ const struct SpeciesInfo gSpeciesInfoGen1[] =
             gOverworldPalette_Rattata,
             gShinyOverworldPalette_Rattata
         )
+        OVERWORLD_FEMALE(
+            sPicTable_RattataF,
```
Comment (hedara90) [inline src/data/pokemon/species_info/gen_1_families.h:1890]: Front female sprite could have visible whiskers, male sprite could have 1 pixel longer whiskers

File: graphics/pokemon/pikachu/overworldf.png
Comment (hedara90) [inline graphics/pokemon/pikachu/overworldf.png:1]: Blinking pixel on north tail sprite

File: graphics/pokemon/gloom/overworldf.png
Comment (hedara90) [inline graphics/pokemon/gloom/overworldf.png:1]: North sprite has blinking pixels on tip of leaves.

**Comment by hedara90** [comment]
Poke me when this is ready

**Comment by kittenchilly** [comment]
On hold as I am currently waiting for a master merge into upcoming

**Comment by kittenchilly** [comment]
Master has been merged and I have updated the branch, so this PR is no longer on hold and can be reviewed

File: graphics/pokemon/sneasel/overworldf.png
Comment (hedara90) [inline graphics/pokemon/sneasel/overworldf.png:1]: Blinking pixels on south sprite.

File: graphics/pokemon/staraptor/overworldf.png
Comment (hedara90) [inline graphics/pokemon/staraptor/overworldf.png:1]: Blinking pixel on east/west on head by beak.

File: graphics/pokemon/wooper/overworldf.png
Comment (hedara90) [inline graphics/pokemon/wooper/overworldf.png:1]: Blinking pixels on south sprite cheeks.

**Review by hedara90 — COMMENTED** [review]
I'll look at the code changes later, only sprite comments right now.

**Comment by hedara90** [comment]
When setting `OW_POKEMON_OBJECT_EVENTS` to `FALSE` compilation fails. This is not the case on regular `upcoming`.
20240930_11h03m42s_grim

File: graphics/pokemon/gloom/overworldf.png
Comment (kittenchilly) [reply graphics/pokemon/gloom/overworldf.png:1]: I don't see this, do you mean the south sprite

File: graphics/pokemon/dodrio/overworldf.png
[the remark this answers is not in the document]
Comment (kittenchilly) [reply graphics/pokemon/dodrio/overworldf.png:1]: I don't see it

File: graphics/pokemon/doduo/overworldf.png
[the remark this answers is not in the document]
Comment (kittenchilly) [reply graphics/pokemon/doduo/overworldf.png:1]: The neck is the same color as the beak, so I don't see the problem there

File: graphics/pokemon/octillery/overworldf.png
[the remark this answers is not in the document]
Comment (kittenchilly) [reply graphics/pokemon/octillery/overworldf.png:1]: Since its not even number of pixels there's not really a way to do it that looks good

**Comment by kittenchilly** [comment]
Addressed the sprite reviews, and fixed the OW_POKEMON_OBJECT_EVENTS compilation fail

File: graphics/pokemon/alakazam/overworldf.png
Comment (hedara90) [inline graphics/pokemon/alakazam/overworldf.png:1]: Still some blinking
20241003_10h48m41s_grim

File: graphics/pokemon/donphan/overworldf.png
Comment (hedara90) [inline graphics/pokemon/donphan/overworldf.png:1]: Still some blinking
20241003_10h59m02s_grim

File: graphics/pokemon/mamoswine/overworldf.png
Comment (hedara90) [inline graphics/pokemon/mamoswine/overworldf.png:1]: Still some blinking
20241003_11h13m44s_grim

File: graphics/pokemon/meganium/overworldf.png
Comment (hedara90) [inline graphics/pokemon/meganium/overworldf.png:1]: Still some blinking
20241003_11h16m09s_grim

File: graphics/pokemon/wooper/overworldf.png
Comment (hedara90) [inline graphics/pokemon/wooper/overworldf.png:1]: Still some blinking
20241003_11h25m35s_grim

**Comment by kittenchilly** [comment]
Addressed

**Review by hedara90 — APPROVED** [review]

**Pull request merged**